### ISSUE TITLE
TEI to JSON conversion skips <hi><w>...</w></hi>

### ISSUE THREAD

**Issue by jjmccollum** [d-flood/Tendon#2]
I noticed that in the IGNTP transcription editor seems to reorder elements when both a rubrication and an editorial expansion are involved. The following example comes from the transcription of GA 0142:

```xml
<hi rend="rubric"><w><ex>και</ex></w></hi>
<w><hi rend="rubric">πιστοις</hi></w>
<w><hi rend="rubric">εν</hi></w>
<w><abbr type="nomSac"><hi rend="overline"><hi rend="rubric">χω</hi></hi></abbr></w>
<w><abbr type="nomSac"><hi rend="overline"><hi rend="rubric">ιυ</hi></hi></abbr></w>
```

The last four words (which involve various degrees of highlighting for _nomina sacra_ and different colors of ink) are encoded in what seems like the most natural way (with everything enclosed in a `<w/>` element), but the first one is not. I would expect

```xml
<w><hi rend="rubric"><ex>και</ex></hi></w>
```

This should be fine for TEI validation, since the `<ex/>` element can be contained by a `<hi/> element (https://tei-c.org/release/doc/tei-p5-doc/en/html/ref-ex.html); I suspect that this is an idiosyncrasy of the transcription editor. That said, this may be a more fitting issue for that tool's GitHub repository, but I wanted to mention it here because Criticus skips the rest of the verse starting at this element, presumably because of the regex pattern.

**Comment by jjmccollum** [comment]
Actually, it looks like the issue is with the handling of `<ex/>` elements. The following line in `from_tei.py` seems to be the issue:

```python
text = re.sub(r'<ex.*</ex>', '', text)
```

I think it might need to be corrected to

```python
text = re.sub(r'<ex[^<>]*>', '', text)
text = re.sub(r'</ex>', '', text)
```

**Comment by d-flood** [comment]
Yes, you're right. We cannot be removing the `<ex>` elements *and* any text or child elements. I've committed the change to this repo and will update the version on PYPI after a few other updates.

**Comment by d-flood** [comment]
This fix (and others) have (hopefully) been addressed in the [v0.33.0 PYPI release](https://pypi.org/project/criticus/0.33.0/)
Don't hesitate to reopen or start a new related issue.
